src/util/check-github-app-type.ts

```diff
@@ -0,0 +1,93 @@
+// this will need to be updated later to check for database entry
+import { Installation } from "models/installation";
+import { GitHubServerApp } from "models/git-hub-server-app";
+import { GitHubInstallationClient } from "../github/client/github-installation-client";
+import { getCloudInstallationId } from "../github/client/installation-id";
+import { booleanFlag, BooleanFlags } from "config/feature-flags";
+import { GitHubUserClient } from "../github/client/github-user-client";
+import Logger from "bunyan"
+import { GitHubAppClient } from "../github/client/github-app-client";
+
+export const GITHUB_ENTERPRISE_CLOUD_HOSTNAME = "https://github.com";
+export const GITHUB_ENTERPRISE_CLOUD_API_BASEURL = "https://api.github.com";
+export const GITHUB_ENTERPRISE_CLOUD_ACCEPT_HEADER = "application/vnd.github.v3+json"
+
+export interface GitHubEnterpriseUrls {
+	hostname: string;
+	apiBaseUrl: string;
+	acceptHeader: string;
+}
+
+export async function getGitHubApiUrl(jiraHost: string) {
+	const gitHubEnterprise = await getGitHubBaseUrl(jiraHost);
+
+	return await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost) && gitHubEnterprise
+		? `${gitHubEnterprise.apiBaseUrl}`
+		: GITHUB_ENTERPRISE_CLOUD_API_BASEURL;
+}
+
+export async function getGitHubHostname(jiraHost: string) {
+	const gitHubEnterprise = await getGitHubBaseUrl(jiraHost);
+
+	return await booleanFlag(BooleanFlags.GHE_SERVER, false, jiraHost) && gitHubEnterprise
+		? gitHubEnterprise.hostname
+		: GITHUB_ENTERPRISE_CLOUD_HOSTNAME;
+}
+
+// TODO: make this function private as soon as all usages have been refactored to one of the factory functions below
```

Comment:
There's no way to make a function private in JS unless it's a method on a class. Is that what you were thinking? Refactor this to a class?